### DISCUSSION TITLE
Disconnecting from Car multimedia system (Android Auto) resets navigation.

### DISCUSSION THREAD

**Discussion by mdykierek** [osmandapp/OsmAnd#14774] [Q&A]
### Description
I am setting a destination. Don't like what I see and want to set an intermediate destination. Pull out USB from my phone and the initial route is gone.
It is even worse, when I already have like 2 intermediate points and want to add one in the middle. Don't know how to do it with car display and all is gone when I take the phone to do it.

### How to reproduce?
Set a destination.
Unplug USB.
Route is gone.

### Your Environment
OsmAnd Version: 4.2.6
Android/iOS version: 11
Device : raphael
Brand : Xiaomi
Model : Mi 9T Pro
Product : raphael_eea
Build : RKQ1.200826.002 test-keys

**Maps used (online or offline):**   
Offline maps.

**Comment by mdykierek** [comment]
Another likely scenario (I had today):
Visit gas station -> turn off the engine.
Start again -> route is gone.

**Comment by vshcherb** [comment]
I'm afraid it belongs to a discussion, probably you can click Previous route. Cause all intermediate and final points are stored and displayed in Previous Route when you click Navigation

**Comment by mdykierek** [comment]
If I return to the car, start the engine I don't seem to have "Previous route" on the car's display (I do have it on the phone). All I have is Favorites and Search icon.
From the safety and usability point of view, it is highly desirable to be able to continue navigating after e.g. engine chokes, refueling, phone plug is not connected right, etc.
Having to play with the phone isn't safe. I may be wrong but AA should improve safety with bigger display, more focused, and less distracting interface.
Doing stuff over is also kind of irritating.
I will definitely play more with it and compare against Maps. I definitely feel like it used to work when I drove 1000km to Italy and back for winter vacation.
When you exit OsmAnd during a trip and start it (the App) again, it asks if you want to resume. Android Auto re-connection doesn't even give that option. I think there is room for improvement here.

One more thing I noticed is that it takes a lot of time for the Android Auto to show the "Start" when a really trivial 5-10 km away destination is set and... Android Auto crashed on me for the first time ever with the "red screen of death".
Another (not related to AA) general observation for the last couple of days - my phone reports high memory use and recommends cleanup. No long trips just usual everyday city rides. Do you have any suggestion how to track this down?

**Reply by vshcherb** [reply]
Unfortunately it's done by Google sending request to Stop Navigation once the phone is disconnected we need to follow that guidlien.

**Reply by mdykierek** [reply]
That's hard to discuss with ;)
Let's think how to make it easier to restore the last route from Android Auto car's display. First item in search or Favorites?
Oh, favorites is another thing that I think can be improved. I think now they are alphabetically sorted. I would find it quite useful to arrange them by distance or frequency of use. One could toggle the modes with subsequent taps?

**Reply by vshcherb** [reply]
You mean to restore it on Android Auto, that's tricky :-( You can restore it on the phone and then connect again to AA.

**Reply by mdykierek** [reply]
I didn't mean automatically. It definitely could be a first item in Search history or extra item in Favorites (last route).

**Comment by mdykierek** [comment]
> One more thing I noticed is that it takes a lot of time for the Android Auto to show the "Start" when a really trivial 5-10 km away destination is set,
Today I noticed 2 reasons:
1. The phone displays the Resume previous route? and waits.
2. The phone doesn't have GPS enabled. Then it never ends. A message/icon showing no GPS/location could help there to figure out what to do (enable GPS, wait to leave a garage or a tunel etc.)
